demos/sev-demo/README.md

```diff
@@ -0,0 +1,209 @@
+    ├── modules.builtin
+    ├── modules.builtin.alias.bin
+    ├── modules.builtin.bin
+    ├── modules.builtin.modinfo
+    ├── modules.dep
+    ├── modules.dep.bin
+    ├── modules.devname
+    ├── modules.order
+    ├── modules.softdep
+    ├── modules.symbols
+    └── modules.symbols.bin
+```
+
+Make sure that the module matches your kernel version. Speaking of which, `efi_secret` also requires some minor kernel patches. These patches and the module are not currently upstream. You can find the v6 patch [here](https://lore.kernel.org/lkml/20211129114251.3741721-1-dovmurik@linux.ibm.com/T/) and a tree containing the v6 patch [here](https://github.com/confidential-containers-demo/linux/tree/conf-comp-secret-v6).
+
+Assuming some x86_64 base config file, you can build a kernel and module like so. 
+```
+cp some-base-config .config
+./scripts/config --enable CONFIG_AMD_MEM_ENCRYPT
+./scripts/config --enable AMD_MEM_ENCRYPT_ACTIVE_BY_DEFAULT
+./scripts/config --enable CONFIG_SECURITYFS
+./scripts/config --enable CONFIG_VIRT_DRIVERS
+./scripts/config --module CONFIG_EFI_SECRET
+make olddefconfig
+
+DIR=/path/to/linux-guest-kernel
+make -j16
+make INSTALL_PATH=$DIR install
+make -j16 INSTALL_MOD_STRIP=1 INSTALL_MOD_PATH=$DIR modules_install
+```
+
+Once you generate an initrd from the updated rootfs, update your Kata configuration file to point to the new initrd. You will also need to update the kernel field in the configuration file to point to your new kernel. For CCv0 you should have already added `agent.config_file=<agent-config-path>` to the `kernel_params` field. 
+
+## Kata Runtime
+
+Since SEV and SEV-ES use pre-attestation driven from outside of the guest, some changes are required for the Kata shim. These changes are not currently upstream. You will need to build your Kata shim with [this PR](https://github.com/kata-containers/kata-containers/pull/3025). 
+You will also need a few changes to `govmm`. For now the easiest way to get them is via [this tree](https://github.com/confidential-containers-demo/govmm/tree/sev-attestation).
+This will work best with QEMU 6.2.0rc0 given that newer version of QEMU will expect to find the `kernel-hashes=on` flag, which is not currently set by the shim. See [here](https://www.mail-archive.com/qemu-devel@nongnu.org/msg851132.html) for details.
+
+You will also need to set the following parameters in the Kata configuration file.
+
+First, you'll need to enable SEV via the `confidential_guest` parameter.
+```
+# Enable confidential guest support.
+# Toggling that setting may trigger different hardware features, ranging
+# from memory encryption to both memory and CPU-state encryption and integrity.
+# The Kata Containers runtime dynamically detects the available feature set and
+# aims at enabling the largest possible one.
+# Default false
+confidential_guest = true
+```
+
+Next you'll need to enable pre-attestation.
+```
+# Enable guest attestation
+# (default: false)
+guest_attestation=true
```

Comment:
Pedantic, but you're referring to "pre-attestation", but the config is for attestation, which could confuse readers.

Reply:
The `guest_attestation` setting enables pre-attestation. This is mentioned a little bit at the beginning of the section. I'll try putting parenthesis around the "pre-". Maybe that makes it slightly clearer.